snarkos/ledger/mod.rs

```diff
@@ -125,6 +115,22 @@ impl<N: Network> Ledger<N> {
         // Return the next block.
         Ok(next_block)
     }
+
+    /// Attempts to add the given block to the ledger.
+    pub(crate) async fn add_next_block(self: &Arc<Self>, next_block: &Block<N>) -> Result<()> {
```

Comment:
very much agree with adding a dedicated method for this operation, was planning on doing that myself :+1: